xbmc/android/activity/XBMCApp.cpp

```diff
@@ -101,6 +102,7 @@ CXBMCApp::CXBMCApp(ANativeActivity* nativeActivity)
   : CJNIApplicationMainActivity(nativeActivity)
   , CJNIBroadcastReceiver("org/xbmc/kodi/XBMCBroadcastReceiver")
 {
+  m_xbmcappinstance = this;
```

Comment:
I never said that singletons aren't ugly (and we don't use any macros or templates for them) but the reason why this is a lot uglier is because nothing stops anyone from creating multiple instances of `CXBMCApp` and then you have absolutely no control over which instance is stored in the static member and things might easily not behave the way you expect it to because some other code created another instance. I know this is not the case with this specific class but from a generic design / implementation point of view that's why a "proper" singleton is less ugly IMO.